tests/ui/weird-exprs.rs

```diff
@@ -231,6 +231,31 @@ fn infcx() {
     let _cx: cx::cx::Cx = cx::cx::cx::cx::cx::Cx;
 }
 
+fn return_already() -> impl std::fmt::Debug {
+    loop {
+        return !!!!!!!
+        break !!!!!!1111
+    }
+}
+
+fn cursed_macros() -> impl std::fmt::Debug {
```

Comment:
This name is misleading. Those are not actually macro invocations, but keywords followed by `!` negations and regular blocks.

Reply:
That's somewhat the whole purpose of `weird-exprs.rs`: Abuse rust's syntax to make an expression look like something different (ensuring that it'll always be parsed correctly). Similarly, `fn closure_matching()` does not in fact match on closures.